modules/adikteevBidAdapter.js

```diff
@@ -1,91 +1,223 @@
+  return {
+    width: window.screen ? window.screen.width : '',
+    height: window.screen ? window.screen.height : '',
+  };
+};
+
+export const getDocumentDimensions = () => {
+  const de = document.documentElement;
+  const be = document.body;
+
+  const bodyHeight = be ? Math.max(be.offsetHeight, be.scrollHeight) : 0;
+
+  const w = Math.max(de.clientWidth, de.offsetWidth, de.scrollWidth);
+  const h = Math.max(
+    de.clientHeight,
+    de.offsetHeight,
+    de.scrollHeight,
+    bodyHeight
+  );
+
+  return {
+    width: isNaN(w) ? '' : w,
+    height: isNaN(h) ? '' : h,
+  };
+};
+
+export const isWebGLEnabled = () => {
+  // Create test canvas
+  let canvas = document.createElement('canvas');
+
+  // The gl context
+  let gl = null;
+
+  // Try to get the regular WebGL
+  try {
+    gl = canvas.getContext('webgl');
+  } catch (ex) {
+    canvas = undefined;
     return false;
   }
-  return sizes.every(size => utils.isArray(size) && size.length === 2);
+
+  // No regular WebGL found
+  if (!gl) {
+    // Try experimental WebGL
+    try {
+      gl = canvas.getContext('experimental-webgl');
+    } catch (ex) {
+      canvas = undefined;
+      return false;
+    }
+  }
+
+  return !!gl;
+};
+
+export const getDeviceInfo = (deviceDimensions, viewDimensions, documentDimensions, webGL) => {
+  return {
+    browserWidth: viewDimensions.width,
+    browserHeight: viewDimensions.height,
+    deviceWidth: deviceDimensions.width,
+    deviceHeight: deviceDimensions.height,
+    documentWidth: documentDimensions.width,
+    documentHeight: documentDimensions.height,
+    webGL: webGL,
+  };
+};
+
+const validateSizes = sizes => utils.isArray(sizes) && sizes.some(size => utils.isArray(size) && size.length === 2);
+
+export function conformCookies(documentCookie) {
+  return documentCookie
+    .split('; ')
+    .filter(i => i.indexOf('=') !== -1)
+    .reduce((acc, kv) => {
+      const [k, v] = kv.split('=');
+      acc[k] = v;
+      return acc;
+    }, {});
 }
 
 export const spec = {
   code: BIDDER_CODE,
   supportedMediaTypes: [BANNER],
 
   isBidRequestValid: (bid) => {
-    setstagingEnvironmentSwitch(stagingEnvironmentSwitch || !!bid.params.stagingEnvironment);
     return !!(
       bid &&
       bid.params &&
-      bid.params.bidFloorPrice &&
       bid.params.placementId &&
       bid.bidder === BIDDER_CODE &&
       validateSizes(bid.mediaTypes.banner.sizes)
     );
   },
 
   buildRequests: (validBidRequests, bidderRequest) => {
-    const payload = {
-      validBidRequests,
-      bidderRequest,
-      refererInfo: bidderRequest.refererInfo,
-      currency: config.getConfig('currency'),
-      userAgent: navigator.userAgent,
-      screen: {
-        width: window.screen.width,
-        height: window.screen.height
+    const payload = Object.assign({},
+      {
+        akPbjsVersion: AK_PBJS_VERSION,
+        bidRequests: validBidRequests.map(conformBidRequest),
+        cookies: conformCookies(document.cookie),
```

Comment:
Why are you sending the document's cookies?

Reply:
I've removed this part.